controllers/gitrepository_controller.go

```diff
@@ -703,13 +709,16 @@ func (r *GitRepositoryReconciler) garbageCollect(ctx context.Context, obj *sourc
 		return nil
 	}
 	if obj.GetArtifact() != nil {
-		if deleted, err := r.Storage.RemoveAllButCurrent(*obj.GetArtifact()); err != nil {
+		deleted, err := r.Storage.RemoveGarbageFiles(*obj.GetArtifact(), r.artifactRetentionRecords, r.artifactRetentionTTL)
```

Comment:
Isn't this competing with other workers? I am not sure if it is worth having the controller workers fighting to garbage collect. We may want to do a leader election or semaphore approach to ensure only a single worker will garbage collect at a given time.

Reply:
I don't think that's going to a problem. Ideally, each worker is going to be reconciling a different source, and thus garbage collecting a different artifact. I might be wrong here. @hiddeco @darkowlzz

Reply:
Every object has its own (metadata based) folder in the Storage, and they're thus not competing.

Reply:
Fair point - closing this accordingly.